### ISSUE TITLE
CleanupOperation (old file removal) needs min/max allowed versions

### ISSUE THREAD

**Issue by binwiederhier** [syncany/syncany#93]
The current implementation of the CleanupOperation removes old file versions if there are more than N versions for a file -- so file histories are only N versions long. Default for N is 5.

The CleanupOperation is called in every UpOperation (unless cleanup is disabled in the UpOperationOptions). If a file "somefile.txt" is updated very often (save, edit, save, edit, save, edit, ...), new file versions for this file are added every time, and if the max. amount of N file versions is reached, the older file versions are deleted.

These deletions are communicated to other clients in PURGE database versions, which is basically a normal database version, but interpreted differently. In the above case, this leads to two database version files (db-clientname-0000X and db-clientname-0000[X+1]) are uploaded every time -- meaning that every time you click "Save", two database files have to be uploaded and then downloaded/processed by the other clients.

Long story short: There should a min. and a max file history length parameter (like in the "merge databases") case. The behavior should be as follows: If a file history is MAX file versions long, delete all but MIN file versions of it. E.g. if it is 15 file versions long, delete all but 5. That would mean that the cleanup must be called only every 10 "up" calls.
